### DISCUSSION TITLE
Is Preventing Consolidation and Centralization an Intent of the TSP?

### DISCUSSION THREAD

**Discussion by mathieuglaude** [trustoverip/trust-spanning-protocol#28] [General]
I want to start by saying that I have very much enjoyed being a listener to the trust spanning protocol (TSP) proposals to-date. So much learning..

From my point-of-view, the conversations to-date have been about what the TSP must support, and not what it’s supposed to prevent/stop.

I read a very interesting recent paper that made me think if what they describe should apply to what we’re doing here. The paper is called: Avoiding Internet Centralization ([here](https://www.ietf.org/archive/id/draft-nottingham-avoiding-internet-centralization-02.html)).

My big question to the working group is: **Is the trust spanning protocol meant to prevent more centralization/consolidation on the Internet?**

Here are some quotes from the above article that spoke to me in the context of the TSP. I’ve also mixed in some of my personal comments.

### Part 1

- _“These difficulties call into question what role architectural design -- in particular, that overseen by open standards bodies such as the IETF -- should play in preventing, mitigating, and controlling consolidation of power on the Internet. This document discusses aspects of centralization that relate to Internet standards efforts, and argues that while the IETF may not be able to prevent these outcomes, there are still meaningful steps it can take to help counteract them.”_
- _“For example, HTTP [HTTP] is not generally considered a centralized protocol; interoperable servers are easy to instantiate, and multiple clients are available. It can be used without central coordination beyond that provided by DNS, as discussed above. However, applications built on top of HTTP (as well as the rest of the “Web Platform”) often exhibit consolidation (for example, social networking). HTTP is therefore an example of platform centralization -- while the protocol itself is not centralized, it facilitates the creation of consolidated services and applications._

My comments:

- Platform centralization seems difficult to prevent with protocol design.
- Even though HTTP isn’t a spanning protocol, protocols like HTTP weren’t able to control what happens on top of them. 
- If we aim to design a protocol to promote decentralization, how do we go about it differently?

### Part 2

- _“Efforts to decentralize often have the effect of merely shifting consolidation to a different place.”_

My comments:

- The parallel for me here is with trust tasks and trust applications.
- Supporting systems such as Indy ledgers, VDRs and TRs can be examples of shifting consolidation. 
- In the article, they describe how governance can “choke” and “block” the system.

### Part 3

-  _Governance layer is a choke point: “when a function requires governance to realize common goals and protect minority interests, a "choke point" is naturally formed by the chosen governance mechanism, increasing centralization.”_
-  _One wouldn’t find out that a protocol is encouraging centralization until it scales “consolidation sometimes only becomes clear after the function is deployed at scale.”_

My comments:

- They’re not saying that consolidation is a bad thing. Through the paper they talk a bunch about DNS and why it’s a good choice for what it does.
- Centralization isn’t a bad thing, but we want our architecture to ensure that our principles are not impacted, that it doesn’t deviate from its original intention over time. 

### Conclusion

While Consolidation and Centralization prevention is clearly outside the ToIP Layer 2 Technical Requirements specified in the Architecture Spec, it may be worth considering including it in the requirements of the TSP. If preventing consolidation and centralization is indeed an intent, then it warrants further exploration. If not, then I still enjoyed reading this paper regardless!

**Comment by talltree** [comment]
@mathieuglaude Thanks for starting a discussion on this very important point. I myself have still not finished the Nottingham paper on [Avoiding Internet Centralization](https://www.ietf.org/archive/id/draft-nottingham-avoiding-internet-centralization-02.html), but what I have read so far is fantastic.

I believe all of your observations about how the ToIP Layer 2 protocol can and cannot help prevent or minimize centralization are right on the money. It is indeed a highly complex topic. I'm looking forward to what other TSPTF members have to say about this topic.

However, regarding question about whether "it may be worth considering including it in the requirements of the TSP", let me point out design principle # 4 in [Design Principles for the ToIP Stack](https://trustoverip.org/permalink/Design-Principles-for-the-ToIP-Stack-V1.0-2022-11-17.pdf):

> **_Decentralization by Design and Default_**
> 
> _To be trusted by all parties, a global network cannot favor any single centralized service or authority; it must allow functionality and authority to be distributed as widely as possible._

So I believe that answers your overall question: decentralization is indeed very clearly a design goal not just of the trust spanning protocol, but of the ToIP stack as a whole.

**Reply by IDmachines** [reply]
Also good reading on the topic particularly wrt DL "Are Blockchains Decentralized" https://blog.trailofbits.com/wp-content/uploads/2022/06/Unintended_Centralities_in_Distributed_Ledgers.pdf
